### ISSUE TITLE
E2506 will fail when using default vpc

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#3000]
### CloudFormation Lint Version

latest

### What operating system are you using?

Mac

### Describe the bug

False positive for rule E2506 when using security group names with a default vpc

### Expected behavior

We should be able to specify a default security group name with a default vpc

### Reproduction template

```yaml
Resources:
  IngressRule:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: 'Security Group Vpc'
      VpcId: "vpc-default-id"
      SecurityGroupIngress:
      -
        IpProtocol: 1
        SourceSecurityGroupName: default
        FromPort: 0
        ToPort: 0

```
